### DISCUSSION TITLE
Cleaning up external offline images

### DISCUSSION THREAD

**Discussion by JackMai123** [immich-app/immich#6341] [Feature Request]
Adding the option up automatically removing offline images post external library scanning. This could streamline storage and ensure data freshness. Thoughts, team?

**Comment by bo0tzz** [comment]
Automatically removing offline assets is risky. What if your NAS goes down and Immich decides that you removed all the files so it can wipe your DB? That seems undesirable.

**Reply by JackMai123** [reply]
A ping can be sent to the Nas to check if the Nas is gone or the files are really deleted. In addition, the deleted file can be rescanned the next time the system is scanning to see if it is really deleted before it delete the thumbnail. It can also be moved to the trash folder as an alternative.
